### PR TITLE
[bitnami/keycloak] add else when '.Values.enableDefaultInitContainers: false', otherwise all initContainers are disabled

### PR DESCRIPTION
**Pull request by lknite**
### Description of the change

Adds an 'else' to the .Values.enableDefaultInitContainers allowing for initContainers to still be specified, even if the default initContainers are disabled.

### Benefits

It is understood that folks sometimes want to specify initContainers and that is good, so we should still allow initContainers to be specified, even if someone decides to disable the default initContainers (needed to run 3rd party images).

### Possible drawbacks

None come to mind.  It's a pretty small change with big benefit.

### Applicable issues

- fixes #24544

### Additional information

I'm not a pro at helm chart template files, so please check my work.  Suggestions welcome!

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### Notes
Been awhile since I commited a change... if something else is needed just let me know.

### CODE REVIEW

**Comment by javsalgar** [comment]
Hi!

Thank you so much for the PR! Could you fix the DCO issue?

**Comment by lknite** [comment]
I canceled the last pr because I didn't add the signature line and used two commits.  Resubmitted this time with one commit and added a signature line.  Thought it was good to go.  When you ask me to fix the DCO issue could you be more specific?  Not sure what you are asking me to do.  (haven't done a lot of commits to open source projects)

**Comment by lknite** [comment]
Oh, it wanted a different signature.  When I read the DCO document it said it wanted my real name and email (which doesn't match my alias here at github.com).  Is it possible for me to use my real name or should I just change it to the alias?

**Comment by lknite** [comment]
DCO is passing check now

File: bitnami/keycloak/templates/statefulset.yaml [hunk 1]
```diff
@@ -116,6 +116,11 @@ spec:
         {{- if .Values.initContainers }}
         {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
         {{- end }}
+      {{- else if and (eq .Values.enableDefaultInitContainers false) (.Values.initContainers) }}
+      initContainers:
+        {{- if .Values.initContainers }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (lknite) [reply bitnami/keycloak/templates/statefulset.yaml:123]: Agreed, your approach is cleaner.

Is there a button that you can click to approve your suggestion, or do I need to update my code with your suggestion and push it again?  (just working through how github does things)

**Review by juan131 — APPROVED** [review]
LGTM!

**Comment by lknite** [comment]
looks like its working now, testing ...

**Comment by juan131** [comment]
Hi @lknite 

Please note that it takes a few hours since the PR is merged until the chart is published in the OCI registry.

**Comment by lknite** [comment]
Pulled 21.0.3 again today and looks like the change isn't there.

**Comment by juan131** [comment]
I'm reporting this issues internally. There must be some issue in our pipeline that's preventing new releases to be published in the OCI registry. I'll keep you updated @lknite

**Comment by juan131** [comment]
Hi @lknite 

It seem it was a cache issue. Could you try again with latest version `21.1.0`?

```console
$ helm pull oci://registry-1.docker.io/bitnamicharts/keycloak
Pulled: registry-1.docker.io/bitnamicharts/keycloak:21.1.0
Digest: sha256:02dea1a303c12eb0c799c3078e9f8c1cab89731a4b4f022937ab9a2cdb4ba20f
```

**Comment by lknite** [comment]
looks good, thank you

**Pull request merged**